youtube_dl/extractor/odnoklassniki.py

```diff
@@ -153,6 +158,10 @@ def _real_extract(self, url):
                 video_id, 'Downloading metadata JSON',
                 data=urlencode_postdata(data))
 
+        paymentInfo = metadata.get('paymentInfo')
+        if paymentInfo:
+            raise ExtractorError('This is Paid video. you need to subscribe in order to watch it', expected=True)
```

Comment:
1. This must only be raised after ensuring the `formats` is empty.
2. Message should be changed to `This video is paid, subscribe to download it`.

Reply:
Thanks. PR was updated.